src/Tizen.Security.DevicePolicyManager/Tizen.Security.DevicePolicyManager/DevicePolicyManager.cs

```diff
@@ -0,0 +1,149 @@
+﻿/*
+ *  Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ */
+
+using System;
+using System.Collections.Generic;
+using Tizen.Internals.Errors;
+
+namespace Tizen.Security.DevicePolicyManager
+{
+    /// <summary>
+    /// The DevicePolicyManager provides the methods to create handle for device policy.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class DevicePolicyManager : IDisposable
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// A constructor of DevicePolicyManager that creates handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <exception cref="InvalidOperationException">Thrown when connection refused or a memory error occurred.</exception>
+        public DevicePolicyManager()
+        {
+            _handle = Interop.DevicePolicyManager.CreateHandle();
+            var lastError = ErrorFacts.GetLastResult();
+            if (lastError != (int)Interop.DevicePolicyManager.ErrorCode.None)
+            {
+                throw DevicePolicyManagerErrorFactory.CreateException(ErrorFacts.GetLastResult());
+            }
+        }
+
+        /// <summary>
+        /// Method to creates an instance of Device Policy.
+        /// </summary>
+        /// <typeparam name="T">The generic type to create.</typeparam>
+        /// <returns>An instance of policy.</returns>
+        /// <since_tizen> 6 </since_tizen>
+        /// <exception cref="InvalidOperationException">Thrown when failed to create instance of the policy.</exception>
+        public T GetPolicy<T>() where T : DevicePolicy
+        {
+            T policy = Activator.CreateInstance(typeof(T),
+                System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance,
+                null, new object[] { this }, null) as T;
+
+            if (policy == null)
```

Comment:
policy can't be `null`, if can't create a object CreateInstace will throw exception.

Reply:
I remove null checking. I added try-catch and rethrow exception. but I can't be sure it is the right solution. I considered below options.
1. remove exception and just return policy
2. add try-catch and rethrow InvalidOperationException
3. add try-catch and return default(T)